### ISSUE TITLE
feat: HuggingFace integration

### ISSUE THREAD

**Issue by thomaspinder** [JaxGaussianProcesses/GPJax#286]
# Feature Request

Demonstrate hoiw a GPJax dataset may be used to hold a HuggingFace dataset.

**Describe Preferred Solution**

A generic method that can coerce a HuggingFace dataset to a GPJax dataset would be ideally. If this is unfeasible, a notebook would also be most welcome.

Tagging @ingmarschuster who has developed a prototype of this.

**Comment by ingmarschuster** [comment]
If we could get Huggingface datasets directly to be Pytrees, that would be the ideal I guess. I haven't found out how to do this (maybe because I am not an expert on Pytrees). However, one can easily "view" huggingface datasets as dictionaries:

```
import datasets as ds

# dummy for demo
dat = ds.Dataset.from_dict({"X": jnp.arange(5), "y": jnp.arange(5)+1})

dat.set_transform(lambda x: x)
dat.set_format("jax")

```
Now `dat[0:3]` and any (automatic) batching by huggingface lib returns dicts like `{'X': Array([0, 1, 2], dtype=int32), 'y': Array([1, 2, 3], dtype=int32)}` and one can take advantage of the underlying super-fast apache parqet/Apache arrow. 

If nobody has ideas on how to make a HF dataset into a pytree directly, would it be an option to represent datasets in GPJax as dicts rather than the current approach?

**Comment by ingmarschuster** [comment]
in that case I'd take it on in a feature branch after holidays...
